### PR TITLE
make use of clientretry.RetryOnConflict in route controller

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

use the common clientretry.RetryOnConflict instead of  ` for i := 0; i < updateNodeStatusMaxRetries; i++ {`

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/route/route_controller.go [hunk 1]
```diff
@@ -177,21 +180,21 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
...
-									UID:       types.UID(nodeName),
-									Namespace: "",
-								}, v1.EventTypeWarning, "FailedToCreateRoute", msg)
-						}
-						glog.Error(msg)
-
-					} else {
-						glog.Infof("Created route for node %s %s with hint %s after %v", nodeName, route.DestinationCIDR, nameHint, time.Now().Sub(startTime))
-						return
+						rc.recorder.Eventf(
```
Comment (wojtek-t) [inline pkg/controller/route/route_controller.go]: Why did you remove the "rc.recorder != nil" if?

Comment (hzxuzhonghu) [reply pkg/controller/route/route_controller.go]: It can never be nil, see pkg/controller/route/route_controller.go#L80

Comment (wojtek-t) [reply pkg/controller/route/route_controller.go]: If you're creating it New method.
But you can create it on your own e.g. for test purposes and then it can potentially be nil.

Please revert this if.

Comment (hzxuzhonghu) [reply pkg/controller/route/route_controller.go]: ok

Comment (hzxuzhonghu) [reply pkg/controller/route/route_controller.go]: @wojtek-t  done

**Comment by wojtek-t** [comment]
@hzxuzhonghu please squash commits and i will approve.

**Comment by hzxuzhonghu** [comment]
done

**Comment by hzxuzhonghu** [comment]
kindly ping @wojtek-t for approval

**Comment by hzxuzhonghu** [comment]
flake
/retest
